### PR TITLE
[FIX] html_editor: traceback on escape to discard image transformation

### PR DESCRIPTION
**Pull request by adch-odoo**
### Steps to Reproduce:

- Open the to-do app
- Add an image (/image, upload an image)
- Select the image
- Click the Transform button
- Press the Escape key, and a traceback occurs

### Desired behavior after PR is merged:

- Clicking the image transform button in `toolbar` clears the image
selection and enables the image transformation.
- Removes the `active` class from the image transformation button.

task-4184571

### CODE REVIEW

File: addons/html_editor/static/src/main/media/image_plugin.js [hunk 1]
```diff
@@ -370,6 +371,8 @@ export class ImagePlugin extends Plugin {
     closeImageTransformation() {
         if (this.isImageTransformationOpen()) {
             registry.category("main_components").remove("ImageTransformation");
+            this.dispatch("CONTENT_UPDATED");
+            this.document.activeElement.blur();
```
Comment (deso-odoo) [inline addons/html_editor/static/src/main/media/image_plugin.js:375]: Why not target same behavior as in 17.0 where using transform on image removes both selection as well as the toolbar?

File: addons/html_editor/static/src/main/media/image_plugin.js [hunk 2]
```diff
@@ -294,7 +294,11 @@ export class ImagePlugin extends Plugin {
         }
     }
 
-    onSelectionChange() {
+    onSelectionChange(selectionData) {
+        const { anchorNode, focusNode } = selectionData.documentSelection;
+        if (!anchorNode && !focusNode) {
+            return;
+        }
```
Comment (dmo-odoo) [inline addons/html_editor/static/src/main/media/image_plugin.js:301]: I don't understand why this change is needed?

Comment (adch-odoo) [reply addons/html_editor/static/src/main/media/image_plugin.js:297]: I made the following changes because removing all ranges of selection triggered the onSelectionChange event after the TRANSFORM_IMAGE action was dispatched, which caused the image transformation to close(https://github.com/odoo/odoo/blob/18.0/addons/html_editor/static/src/main/media/image_plugin.js#L294). 

Also after a rebase my fix was not working due to a recent commit (https://github.com/odoo/odoo/commit/62be82dd2c6660ef075073d61b560dfd60220c93), the behavior has changed: after dispatching the action, if the selection is no longer within the editable area, it sets the selection back into the editable. As a result, anchorNode and focusNode are no longer null, and in onSelectionChange, this causes the image transformation to close. 

One possible fix is to remove the range in the next tick.

**Comment by dmo-odoo** [comment]
@robodoo r+
